src/components/ModalOpen.tsx

```diff
@@ -217,10 +244,16 @@ class ModalOpenInternal extends React.Component<ModalOpenInternalProps, ModalOpe
             <h1>{t("Open local Style")}</h1>
             <p>{t("Open a local JSON style from your computer.")}</p>
             <div>
-              <InputButton
-                className="maputnik-big-button"
-                onClick={this.onOpenFile}><MdFileUpload/> {t("Open Style")}
-              </InputButton>
+              {typeof window.showOpenFilePicker === "function" ? (
```

Comment:
Can we use if-else here? I think it will be clearer...

Reply:
Is this the syntax you want to go with?

```jsx
  <div>
      {(() => {
        if (typeof window.showOpenFilePicker === "function") {
          return (
            <InputButton
                  className="maputnik-big-button"
                  onClick={this.onOpenFile}><MdFileUpload/> {t("Open Style")}
                </InputButton>
          )
        } else {
          return (
            <FileReaderInput onChange={this.onFileChanged} tabIndex={-1} aria-label={t("Open Style")}>
                  <InputButton className="maputnik-upload-button"><MdFileUpload /> {t("Open Style")}</InputButton>
                </FileReaderInput>
          )
        }
      })()}
    </div>
```

Reply:
Is the lambda mandatory? I recall that react has a simpler way to add if inside the render method...?

Reply:
Not sure how? I can't find anything about if in the [react docs](https://legacy.reactjs.org/docs/conditional-rendering.html) nor [stackoverflow](https://stackoverflow.com/questions/40477245/is-it-possible-to-use-if-else-statement-in-react-render-function). Using a tenary operator is the shortest way. Alternatively, we could move the html block before the return statement and use it from a variable.